### PR TITLE
[doc] add a doc about adoption metrics

### PR DESCRIPTION
**Pull request by manhhung741**
I'd like to add a doc explaining the 2 metrics Sorbet team recommend to use to track adoption progress: file-level & input sends level coverage. Currently, I add this doc as a new item in the sidebar, but I can see it being part of an existing page ("adopting sorbet"?)

Please feel free to give feedback on the writing or the organization of the content (and let me know if I understand the metrics correctly)

### Motivation

It's important to understand the adoption metrics and use them tactically to drive the adoption. I've found them valuable and I think we should have documentation about them.

Knowing this metrics should help developers tracking and communicate adoption progress to their team.

### Test plan

I looked at https://github.com/sorbet/sorbet/blob/master/website/README.md and followed the guide but I don't know how to preview the changes locally to include a screenshot here.

### CODE REVIEW

File: website/docs/tracking_adoption.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
+id: tracking_adoption
+title: Measuring Type-checking Coverage
+sidebar_label: Tracking Coverage
+---
+
+There are two primary metrics that we recommend to track the adoption at Stripe:
```
Comment (jez) [inline website/docs/tracking_adoption.md]:
```suggestion
There are two primary metrics that we recommend to track adoption of Sorbet in a codebase:
```

File: website/docs/tracking_adoption.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+- File-level coverage: the number of files in each type
```
Comment (jez) [inline website/docs/tracking_adoption.md]:
```suggestion
- File-level coverage: the number of files at each
```

File: website/docs/tracking_adoption.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  ...
+  {
+   "name": "example.types.input.sends.typed",
+   "value": 69921
+  },
+  {
+   "name": "example.types.input.sends.total",
+   "value": 120061
+  },
+}
```
Comment (jez) [inline website/docs/tracking_adoption.md]: Do you need to `cat test.txt` for this to show up as output?

File: website/docs/tracking_adoption.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+- File-level coverage: the number of files in each type
+  [strictness level](static.md). There are 5 strictness levels
+- Input send coverage: the number of method calls called on a typed object. Only
+  reported in type-checked files (strictness level true or higher)
+
+Together, they represent how much of a codebase is type-checked by sorbet.
+
+```
+$ srb tc --metrics-file test.txt --metrics-prefix example
```
Comment (jez) [inline website/docs/tracking_adoption.md]:
```suggestion
$ srb tc --metrics-file metrics.json --metrics-prefix example
```

File: website/docs/tracking_adoption.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  },
+  {
+   "name": "example.types.input.sends.total",
+   "value": 120061
+  },
+}
+```
+
+The metrics are important at different phases of adoption. During the ramp-up,
+the first metric is more important. During this phase, we should focus on
```
Comment (jez) [inline website/docs/tracking_adoption.md]: Might be worth repeating that "the first metric" means "file-level typedness"

**Comment by jez** [comment]
> I looked at https://github.com/sorbet/sorbet/blob/master/website/README.md and followed the guide but I don't know how to preview the changes locally to include a screenshot here.

To preview changes you can do

```
cd website
yarn install
yarn start
```

and it should automatically open on localhost in your browser.

**Comment by jez** [comment]
I really liked your idea of a doc. I took some time to flesh it out into a bit longer of a doc so that it has more information, and pushed it to your branch. Thanks again! I'll merge this when the tests pass.

**Comment by manhhung741** [comment]
Thank you, Jez! Sorry I didn’t have time to respond to your feedback
earlier!

**Comment by ghiculescu** [comment]
today i learned about https://sorbet.org/docs/metrics#reporting-metrics-directly-to-statsd, that's pretty cool. is it worth documenting *what* metrics are sent to statsd @manhhung741 ? (i guess i could test it myself but it might be a useful thing to add to the docs)

**Comment by ghiculescu** [comment]
oh also. might be a better question for @paracycle but in https://www.youtube.com/watch?v=v9oYeSZGkUw -  is this graph based on the call-site-level typedness metrics or on something else:

image

**Comment by paracycle** [comment]
@ghiculescu that graph was just counting the number of times developers at Shopify had invoked `srb tc` on our monolith. We similarly track usage via StatsD metrics, and we can also identify which unique users have invoked type-checking. Each color in a bar is a unique user that invoked type checking on that day and the height of the bars are the total number of times it was invoked overall.

The nice thing with graphing those metrics is that you can see a clear trend of people using type checking as part of their development workflows, which to us signals that they are getting quick and early feedback about their code changes. You can clearly see the total invocations increasing starting mid-August and the colors being more varied in each bar.

@Morriar was the person who put that graph together and can talk more about it if you have further questions.

**Pull request merged**